libraries/c_sdk/standard/ble/test/iot_mqtt_ble_system_test.c

```diff
@@ -0,0 +1,1034 @@
+ */
+static bool receivedPubAck = false;
+
+/**
+ * @brief Represents incoming PUBLISH information.
+ */
+static MQTTPublishInfo_t incomingInfo;
+
+/**
+ * @brief Buffer to store incoming publish topic name for assertions in tests.
+ */
+static uint8_t incomingTopicBuffer[ TEST_MQTT_LWT_TOPIC_LENGTH ];
+
+/**
+ * @brief Buffer to store incoming PUBLUSH payload for assertions in tests.
+ */
+static uint8_t incomingPayloadBuffer[ sizeof( MQTT_EXAMPLE_MESSAGE ) ];
+
+/**
+ * @brief Disconnect when receiving this packet type. Used for session
+ * restoration tests.
+ */
+static uint8_t packetTypeForDisconnection = MQTT_PACKET_TYPE_INVALID;
+
+/**
+ * @brief Global entry time to use as a reference timestamp in the getTimeMs()
+ * function. getTimeMs() function will always return the difference between the
+ * current time and the global entry time. This reduces the chances of overflow
+ * for the 32 bit unsigned integer used for holding the timestamp.
+ */
+static uint32_t globalEntryTime;
+
+/**
+ * @brief Flag to represent if BLE is connected.
+ */
+static bool isBleConnected;
+
+/**
+ * @brief Semaphore used to synchronize with BLE connection callback.
+ */
+static SemaphoreHandle_t bleConnectionSem;
+
+/**
+ * @brief Semaphore used to synchronize with BLE channel callback.
+ */
+static SemaphoreHandle_t bleChannelSem;
+
+/**
+ * @brief The timer query function provided to the MQTT context.
+ *
+ * @return Time in milliseconds.
+ */
+static uint32_t getTimeMs();
+
+/**
+ * @brief Sends an MQTT CONNECT packet over connected BLE Channel.
+ *
+ * @param[in] pContext MQTT context pointer.
+ * @param[in] pNetworkContext Network context for OpenSSL transport implementation.
+ * @param[in] createCleanSession Creates a new MQTT session if true.
+ * If false, tries to establish the existing session if there was session
+ * already present in broker.
+ */
+static void establishMqttSession( MQTTContext_t * pContext,
+                                  NetworkContext_t * pNetworkContext,
+                                  bool createCleanSession );
+
+/**
+ * @brief Handler for incoming acknowledgement packets from the broker.
+ * @param[in] pPacketInfo Info for the incoming acknowledgement packet.
+ * @param[in] packetIdentifier The ID of the incoming packet.
+ */
+static void handleAckEvents( MQTTPacketInfo_t * pPacketInfo,
+                             uint16_t packetIdentifier );
+
+/**
+ * @brief The application callback function that is expected to be invoked by the
+ * MQTT library for incoming publish and incoming acks received over the network.
+ *
+ * @param[in] pContext MQTT context pointer.
+ * @param[in] pPacketInfo Packet Info pointer for the incoming packet.
+ * @param[in] pDeserializedInfo Deserialized information from the incoming packet.
+ */
+static void eventCallback( MQTTContext_t * pContext,
+                           MQTTPacketInfo_t * pPacketInfo,
+                           MQTTDeserializedInfo_t * pDeserializedInfo );
+
+/*
```

Comment:
Next few function docs are not in correct doxygen format.

Reply:
Fixed.